dotnet/src/webdriver/RelativeBy.cs

```diff
@@ -108,8 +108,28 @@ public override ReadOnlyCollection<IWebElement> FindElements(ISearchContext cont
             filterParameters["filters"] = this.filters;
             parameters["relative"] = filterParameters;
             object rawElements = js.ExecuteScript(wrappedAtom, parameters);
-            ReadOnlyCollection<IWebElement> elements = rawElements as ReadOnlyCollection<IWebElement>;
-            return elements;
+
+            if (rawElements is ReadOnlyCollection<IWebElement> elements)
```

Comment:
Sorry, I didn't dive deeper here, seems it would be better to rely on some lower interface like `IReadOnlyCollection<...>`? Leaving this comment just to pay an attention what exactly `.ExecuteScript()` returns in terms of types - json serialization topic again.

Reply:
We would have to use a non-generic interface like `ICollection` or `IEnumerable` to capture the `ReadOnlyCollection<object>`. At that point we would need to do something like `.OfType<object>().Cast<IWebElement>()` or something like that.

This is a band-aid for the real need of a generic `.ExecuteScript()` method

Reply:
If you look at the `WebDriver.ParseJavaScriptReturnValue` method, you will see that because the de-serializer tries to take a guess about the return type. This is because it has no type information, which would require a generic method.

Reply:
The generic method is obvious: it is only one way how an user is able to specify the expected/required type (json serializable?). For reviewers: we should understand why `rawElements is ReadOnlyCollection<IWebElement>` and no more..

Reply:
Whenever there is more than one element in the response, the de-serializer will recognize that we are returning elements and de-serialize to a `ReadOnlyCollection<IWebElement>`.

However, there is a special case when no elements are returned: the response is `[]` and the de-serializer will not know we are expecting elements. That is why it returns a `ReadOnlyCollection<IWebElement>`, and that is the scenario we are handling here.

Reply:
Now I see, that is generic problem that all commands in Selenium are not strongly-typed. Thanks for diving into this particular issue.